### PR TITLE
Merge release/3.5 to release/3.6

### PR DESCRIPTION
**Pull request by anwar6953**
Merge release/3.5 to release/3.6

There were conflicts in the following files:
cdap-docs/admin-manual/build.sh
cdap-docs/build.sh
cdap-docs/reference-manual/build.sh
cdap-docs/reference-manual/source/release-notes.rst

http://builds.cask.co/browse/CDAP-RUT318-1

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java [hunk 1]
```diff
@@ -40,10 +43,15 @@
...
 
-  private static final Gson GSON =
-    new GsonBuilder().registerTypeAdapter(PartitionFilter.Condition.class, new ConditionCodec()).create();
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(PartitionFilter.Condition.class, new ConditionCodec())
+    .registerTypeAdapter(PartitionKey.class, new PartitionKeyCodec())
+    .create();
   private static final Type PARTITION_FILTER_LIST_TYPE = new TypeToken<List<PartitionFilter>>() { }.getType();
+  private static final Type PARTITION_KEY_LIST_TYPE = new TypeToken<List<PartitionKey>>() { }.getType();
+
```
Comment (ecfm) [inline cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java:54]: Remove empty line?

File: cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceTaskContext.java [hunk 2]
```diff
@@ -16,11 +14,15 @@
...
 
+package co.cask.cdap.api.mapreduce;
+
 import co.cask.cdap.api.RuntimeContext;
 import co.cask.cdap.api.ServiceDiscoverer;
 import co.cask.cdap.api.TaskLocalizationContext;
 import co.cask.cdap.api.annotation.Beta;
 import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.data.batch.InputContext;
+import co.cask.cdap.api.data.batch.PartitionedFileSetInputContext;
```
Comment (maochf) [inline cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceTaskContext.java]: Unused import?

File: cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java (hunk 1 above)
Comment (anwar6953) [reply cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java:54]: Done

**Comment by awholegunch** [comment]
`cdap-docs` files LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/input/TaggedInputSplit.java [hunk 3]
```diff
@@ -35,94 +32,56 @@
...
   }
 
-  /**
-   * Retrieves the name of this TaggedInputSplit.
-   *
-   * @return name of the TaggedInputSplit
-   */
-  public String getName() {
-    return name;
-  }
```
Comment (ecfm) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/input/TaggedInputSplit.java]: extra empty line?

File: cdap-common/src/main/resources/cdap-default.xml [hunk 4]
```diff
@@ -881,6 +881,22 @@
...
+      (set by ${kafka.zookeeper.namespace}) when setting up a connection to
+      the Kafka service used by CDAP. If the same Kafka service ZooKeeper quorum
+      and namespace are shared by multiple CDAP instances, each CDAP instance
+      needs to distinguish its Kafka topics from those of other CDAP instances
+      with unique values for ${audit.kafka.topic}, ${log.kafka.topic},
+      ${notification.kafka.topic}, ${metadata.updates.kafka.topic}, and
+      ${metrics.kafka.topic.prefix}.
+    </description>
+  </property>
+
```
Comment (ecfm) [inline cdap-common/src/main/resources/cdap-default.xml:899]: extra empty line

Comment (awholegunch) [reply cdap-common/src/main/resources/cdap-default.xml:899]: No, we've been putting two blank lines before the comments that mark the start of a configuration section. The exception is the first one, as it does not require separating from what precedes it...

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**